js/functions.js

```diff
@@ -103,7 +103,9 @@
 
 	// Add a class to big image and caption larger than or equal to 840px.
 	function bigImageClass() {
-		if ( $body.hasClass( 'page' ) || $body.hasClass( 'search' ) || $body.hasClass( 'single-attachment' ) || $body.hasClass( 'error404' ) ) {
```

Comment:
That's correct, should you decide to keep the check in JavaScript some quick perf tests show `.is` will be faster than `.hasClass` in most instances.

https://jsbin.com/facejo/edit?html,js,output

I'd still prefer to do the conditional checks in PHP as there's no way to be sure what a developer will do to `body_class` in a child theme.